src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever.java

```diff
@@ -4,245 +4,243 @@
+                .build();
+
+            logger.log(Level.FINE, "Retrieve logs for traceId: " + traceId);
+            SearchResponse<ObjectNode> searchResponse = this.esClient.search(searchRequest, ObjectNode.class);
+            hits = searchResponse.hits().hits();
+            span.setAttribute("results", hits.size());
+            completed = hits.size() != PAGE_SIZE; // TODO is there smarter?
+
+            if (completed) {
+                logger.log(Level.FINE, () -> "Clear scrollId: " + pitId + " for trace: " + traceId + ", span: " + spanId);
+                esClient.closePointInTime(p -> p.id(pitId));
             }
-            ConsoleNotes.write(writer, message, annotations);
-        }
-    }
 
-    /**
-     * @return the RestClientBuilder to create the Elasticsearch REST client.
-     */
-    private RestClientBuilder getBuilder() {
-        RestClientBuilder builder = RestClient.builder(HttpHost.create(this.elasticsearchUrl));
-        builder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
-        return builder;
-    }
+            ByteBuffer byteBuffer = new ByteBuffer();
 
-    /**
-     * Search the log lines of a build ID and Node ID.
-     *
-     * @param traceId build ID to search for the logs.
-     * @param spanId  A page with log lines the results of the search.
-     * @return A page with log lines the results of the search. The object contains the scrollID to use in {@link #next(String)} requests.
-     * @throws IOException
-     */
-    public SearchResponse search(@Nonnull String traceId, @CheckForNull String spanId) throws IOException {
-        try (RestHighLevelClient client = new RestHighLevelClient(getBuilder())) {
-            final Scroll scroll = new Scroll(DEFAULT_TIMEVALUE);
-            SearchRequest searchRequest = new SearchRequest(this.indexPattern);
-            searchRequest.scroll(scroll);
-            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-            searchSourceBuilder.size(PAGE_SIZE);
-            searchSourceBuilder.sort(new FieldSortBuilder(TIMESTAMP).order(SortOrder.ASC));
-            if (StringUtils.isBlank(spanId)) {
-                searchSourceBuilder.query(matchQuery(SPAN_ID, traceId));
-            } else {
-                searchSourceBuilder.query(boolQuery().must(matchQuery(TRACE_ID, traceId)).must(matchQuery(SPAN_ID, spanId)));
+            try (Writer w = new OutputStreamWriter(byteBuffer, charset)) {
+                String logsVisualizationUrl = kibanaBaseUrl + "/app/logs/stream?" +
+                    "logPosition=(end:now,start:now-1d,streamLive:!f)&" +
+                    "logFilter=(language:kuery,query:%27trace.id:" + traceId + "%27)&";
+                if (pageNo == 0) {
+                    w.write("View logs in Kibana: " + logsVisualizationUrl + "\n\n");
```

Comment:
@kuisathaverat do you know how to produce an HTML link? I guess it's related to the code of io.jenkins.plugins.opentelemetry.job.log.ConsoleNotes

Reply:
it depends on the render you have configured for Jenkins, my experience on showing an URL in the console log is that will work if the URL is simple, but in the case of the Kibana URL is not simple, it has many parameters with delimiters and so on, in my test I could make the link work in all the cases (probably URL encoded works better, but I did not test it), and the link is ugly because is too long. Because of that, I changed to use an Action in the previous plugin.